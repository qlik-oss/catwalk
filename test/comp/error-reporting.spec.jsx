import React from 'react';
import ReactGA from 'react-ga';
import renderer from 'react-test-renderer';

import UseErrorReporting from '../../src/components/error-reporting';

describe('error-reporting', () => {
  it('should return ´null´ if GA not set', () => {
    const tree = renderer.create(<UseErrorReporting GA={null} />).toJSON();

    expect(tree).toMatchSnapshot();
    global.window.close();
  });

  it('should display cookie if not set', () => {
    const sandbox = sinon.createSandbox();
    sandbox.stub(ReactGA, 'initialize');
    sandbox.stub(ReactGA, 'pageview');
    sandbox.stub(window, 'location').value({ pathname: 'some value' });
    sandbox.stub(process, 'env').value({ GA: 'some value' });

    const tree = renderer.create(<UseErrorReporting GA={process.env.GA} />).toJSON();

    expect(tree).toMatchSnapshot();
    sandbox.restore();
  });
});
