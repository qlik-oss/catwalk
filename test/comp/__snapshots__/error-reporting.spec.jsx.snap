// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'should display cookie if not set' 1`] = `
<div
  className="cookieConsent "
  style={
    Object {
      "alignItems": "baseline",
      "background": "#5F6062",
      "bottom": "0",
      "color": "#D6D6D6",
      "display": "flex",
      "flexWrap": "wrap",
      "fontSize": "2rem",
      "justifyContent": "space-between",
      "left": "0",
      "position": "fixed",
      "width": "100%",
      "zIndex": "999",
    }
  }
>
  <div
    className=""
    style={
      Object {
        "flex": "1 0 300px",
        "margin": "15px",
      }
    }
  >
    This website uses cookies to ensure you get the best experience on our website.
     
    <a
      href="https://www.qlik.com/us/legal/cookies-and-privacy-policy"
      rel="noopener noreferrer"
      style={
        Object {
          "color": "#D6D6D6",
        }
      }
      target="_blank"
    >
      Learn more
    </a>
  </div>
  <button
    className=""
    id=""
    onClick={[Function]}
    style={
      Object {
        "background": "#3D8706",
        "border": "0",
        "borderRadius": "0px",
        "boxShadow": "none",
        "color": "#fff",
        "cursor": "pointer",
        "flex": "0 0 auto",
        "fontSize": "2rem",
        "margin": "15px",
        "padding": "5px 10px",
      }
    }
  >
    Got it!
  </button>
</div>
`;
